Sources/MapboxMaps/Snapshot/Snapshotter.swift

```diff
@@ -32,6 +32,7 @@ public class Snapshotter {
         self.options = options
         mapSnapshotter = MapSnapshotter(options: MapboxCoreMaps.MapSnapshotOptions(options))
         style = Style(with: mapSnapshotter)
+        EventsManager(accessToken: options.resourceOptions.accessToken).sendTurnstile()
```

Comment:
This is the important change. This approach matches [Android](https://github.com/mapbox/mapbox-maps-android/pull/920), but there are a few questions we need to think about:

* EventsManager is a thin wrapper around `MMEEventsManager.shared()`. Each time an `EventsManager` is initialized, it will call `MMEEventsManager.shared().initialize(…)`. Is it okay for that to happen more than once (will happen with each snapshotter and map view that is created). What if you are using different access tokens for different purposes in the same app?
    * Based on https://github.com/mapbox/mapbox-events-ios/blob/main/Sources/MapboxMobileEvents/MMEEventsManager.m#L92-L101, it seems like subsequent calls to initialize will just update the access token, but won't repeat any other setup.
* One problem with having more than one instance of `EventsManager` is that each instance will observe changes to the `"MGLMapboxMetricsEnabled"` user defaults key and will call `flush` upon receiving a memory warning. These behaviors only need to happen once per `MMEEventsManager` instance.
    * This MME design means we need to convert EventsManager to a singleton as well to avoid doing more work than necessary. @vesh93 I'd love to see us move away from using a singleton pattern. It's a bad match for our initialization APIs which would lead you to believe there's nothing stopping you from using multiple access tokens simultaneously in the same process.

cc: @vesh93

Reply:
Updated to convert EventsManager to a singleton.